### PR TITLE
Add Patch::Estimate methods for GLEC-certified shipping estimates

### PR DESCRIPTION
**Pull request by holtbp**
### What
- Adds `Patch::Estimate.create_air_shipping_estimate` method
- Adds `Patch::Estimate.create_rail_shipping_estimate` method
- Adds `Patch::Estimate.create_road_shipping_estimate` method
- Adds `Patch::Estimate.create_sea_shipping_estimate` method

### Why
To provide GLEC-certified shipping estimates!

### SDK Release Checklist

- [ ] Have you added an integration test for the changes?
- [ ] Have you built the gem locally and made queries against it successfully?
- [ ] Did you update the changelog?
- [ ] Did you bump the package version [in the code generator](https://github.com/patch-technology/client-code-generation/blob/main/configs/ruby-config.json#L11-L12)?
- [ ] If endpoints were removed, did you manually remove the corresponding files? (this should be rare)
- [ ] For breaking changes, did you plan for the release of the new SDK versions and deploy the API to production?

### CODE REVIEW

File: lib/patch_ruby/models/create_order_request.rb [hunk 1]
```diff
@@ -232,7 +232,7 @@ def valid?
       return false if !@mass_g.nil? && @mass_g > 100000000000
       return false if !@mass_g.nil? && @mass_g < 0
       return false if !@total_price_cents_usd.nil? && @total_price_cents_usd < 1
-      state_validator = EnumAttributeValidator.new('String', ["draft", "placed"])
+      state_validator = EnumAttributeValidator.new('String', ["draft", "reserved", "placed"])
```
Comment (holtbp) [inline lib/patch_ruby/models/create_order_request.rb:235]: @joustrich Is introducing this okay?

**Review by kleinjm — APPROVED** [review]

File: lib/patch_ruby/models/create_order_request.rb (hunk 1 above)
Comment (pcothenet) [reply lib/patch_ruby/models/create_order_request.rb:235]: On behalf of @joustrich, I'll say yes 😁

**Comment by pcothenet** [comment]
@holtbp don't forget a PR for your changes to https://github.com/patch-technology/client-code-generation too!

**Review by pcothenet — APPROVED** [review]

File: lib/patch_ruby/models/create_order_request.rb (hunk 1 above)
Comment (joustrich) [reply lib/patch_ruby/models/create_order_request.rb:235]: Shoot sorry I totally missed this. Yes we do want this in there!

File: lib/patch_ruby/api/orders_api.rb [hunk 2]
```diff
@@ -166,9 +165,10 @@ def create_order_with_http_info(create_order_request, opts = {})
     # @param [Hash] opts the optional parameters
     # @option opts [PlaceOrderRequest] :place_order_request
     # @return [OrderResponse]
-    def place_order(id, place_order_request = {}, opts = {})
-      _place_order_request = Patch::PlaceOrderRequest.new(place_order_request)
-      data, _status_code, _headers = place_order_with_http_info(id, opts.merge!({place_order_request: place_order_request}))
+    def place_order(order_id, opts = {})
+      place_order_request = Patch::PlaceOrderRequest.new(opts)
+      opts[:place_order_request] = place_order_request
+      data, _status_code, _headers = place_order_with_http_info(order_id, opts)
```
Comment (pcothenet) [inline lib/patch_ruby/api/orders_api.rb:171]: @holtbp finding out after the fact, but it looks like it's a change you generated manually, is it?

**Pull request merged**